hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java

```diff
@@ -678,23 +679,50 @@ public boolean isDereferencedBySubclassProperty() {
 
 	// ParameterContainer impl ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 	private List<ParameterSpecification> embeddedParameters;
+	private List<ParameterSpecification> embeddedFilterParameters;
```

Comment:
I think it's much easier if we instantiate the Lists here and then we can add entries without checking for null. The same goes when we check if the list is empty since we don;t have to check for null there as well.